server/src/main/java/org/opensearch/index/cache/request/ShardRequestCache.java

```diff
@@ -45,30 +48,43 @@
 @PublicApi(since = "1.0.0")
 public final class ShardRequestCache {
```

Comment:
1. We cannot change the API of this class as it is `@PublicApi` and needs to maintain bwc.
2. I think we can keep this separate and deprecate over time and implement a new `RequestCacheStats` and `ShardRequestCache` class type which works with storeType info. Also instead of creating the statsHolder for each storeType, we should create only for the ones which is enabled. But this change should again be part of separate PR and for now you can have assertions in `AbstractIndexShardCacheEntity` that `cacheStoreType` is always `ON_HEAP`

Reply:
Makes sense. Let me do that.

Reply:
I have removed changes made to ShardRequestCache for now. Not required. Will add once I integrate IndicesRequestCache with TieredCache as part of separate PR.

Regarding 
> We cannot change the API of this class as it is @PublicApi and needs to maintain bwc.

I guess we can still make changes to existing class and using Version guard checks we ensure BWC. Let me know if my understanding is correct here.

Reply:
Missed attaching the link in last comment. See [here](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#developer-api). Here API is the methods, so there is no version check as such if you change method signature. Or may be I am not understanding your comment correctly

Reply:
I was about talking something like this - https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/tasks/TaskInfo.java#L217-L219

It is possible to change class with @PublicApi. Like example above, we introduce version guard checks where we write/read by checking desired OpenSearch versions. Doing this, it is possible to maintain BWC.

So I think we can still reuse ShardRequestCache and introduce these changes.